R/vroom_write.R

```diff
@@ -58,8 +58,9 @@ vroom_write <- function(x, file, delim = '\t', eol = "\n", na = "NA", col_names
   # Standardise path returns a list, but we will only ever have 1 output file.
   file <- standardise_one_path(file, write = TRUE)
 
-  # If there are no columns in the data frame, just create an empty file and return
-  if (NCOL(x) == 0) {
+  # If there are no columns in the data frame and we aren't appending
+  # just create an empty file and return
+  if (NCOL(x) == 0 && !append) {
```

Comment:
What about the case where `NCOL(x) == 0 && append`?
It feels like that should still be an early return.

Reply:
Notes: Since we weren't returning early when `NCOL(x) == 0 && append` (ie writing an empty dataframe to a file with data already present), `vroom_write_` was being passed an empty data frame, which caused R to crash on Windows with R 4.1 and 3.6 (both of which are 32-bit builds).